crypto/evp/pmeth_gn.c

```diff
@@ -247,14 +251,16 @@ const OSSL_PARAM *EVP_PKEY_param_fromdata_settable(EVP_PKEY_CTX *ctx)
 {
     /* We call fromdata_init to get ctx->keymgmt populated */
     if (fromdata_init(ctx, EVP_PKEY_OP_UNDEFINED))
-        return evp_keymgmt_importdomparam_types(ctx->keymgmt);
+        return evp_keymgmt_import_types(ctx->keymgmt,
+                                        OSSL_KEYMGMT_SELECT_ALL_PARAMETERS);
     return NULL;
 }
 
 const OSSL_PARAM *EVP_PKEY_key_fromdata_settable(EVP_PKEY_CTX *ctx)
 {
     /* We call fromdata_init to get ctx->keymgmt populated */
     if (fromdata_init(ctx, EVP_PKEY_OP_UNDEFINED))
-        return evp_keymgmt_importdomparam_types(ctx->keymgmt);
+        return evp_keymgmt_import_types(ctx->keymgmt,
+                                        OSSL_KEYMGMT_SELECT_KEYPAIR);
```

Comment:
should this be SELECT_ALL ?

Reply:
This is `EVP_PKEY_key_fromdata_settable`, note **`key`** as opposed to **`param`** in `EVP_PKEY_param_fromdata_settable`.

If we collapse the `fromdata` functions to not distinguish between params and the key itself, then I can definitely see ussing SELECT_ALL.  That's for another PR, though...